Kernel/Bus/USB/USBConfiguration.cpp

```diff
@@ -50,8 +54,46 @@ ErrorOr<void> USBConfiguration::enumerate_interfaces()
             // FIXME: It looks like HID descriptors come BEFORE the endpoint descriptors for a HID device, so we should load
             // these too eventually.
             // See here: https://www.usb.org/defined-class-codes
-            if (interface_descriptor->interface_class_code == USB_CLASS_HID)
+            if (interface_descriptor->interface_class_code == USB_CLASS_HID) {
+                USBEndpointDescriptor endpoint_descriptor;
+                memcpy(&endpoint_descriptor, raw_endpoint_descriptor_offset, sizeof(USBEndpointDescriptor));
+
+                if constexpr (USB_DEBUG) {
+                    dbgln("Endpoint Descriptor {}", endpoint);
+                    dbgln("Endpoint Address: {}", endpoint_descriptor.endpoint_address);
+                    dbgln("Endpoint Attribute Bitmap: {:08b}", endpoint_descriptor.endpoint_attributes_bitmap);
+                    dbgln("Endpoint Maximum Packet Size: {}", endpoint_descriptor.max_packet_size);
+                    dbgln("Endpoint Poll Interval (in frames): {}", endpoint_descriptor.poll_interval_in_frames);
+                }
+
+                // FIXME: Allow to set different configurations...
+                USBConfigurationDescriptor descriptor;
+                memcpy(&descriptor, &m_descriptor, sizeof(USBConfigurationDescriptor));
+                transfer_length = TRY(m_device.control_transfer(USB_REQUEST_TRANSFER_DIRECTION_DEVICE_TO_HOST, USB_REQUEST_SET_CONFIGURATION, 1, 0, sizeof(USBConfigurationDescriptor), &descriptor));
+                if constexpr (USB_DEBUG) {
+                    dbgln("USB device configuration was set!");
+                }
+
+                // FIXME: Detect USB HID mouse in an abstracted way...
+                if (interface_descriptor->interface_sub_class_code == 0x1 && interface_descriptor->interface_protocol == 0x2) {
+                    auto interrupt_in_pipe = TRY(USB::InterruptInPipe::create(device().controller(), endpoint_descriptor.endpoint_address, endpoint_descriptor.max_packet_size, device().address(), 10));
+                    m_device.set_interrupt_in_pipe({}, move(interrupt_in_pipe));
```

Comment:
I think we should store the `InterruptInPipe` in the `USB{Mouse,Keyboard}Device`, instead of storing it into the `USBDevice`.